core/controllers/reader.py

```diff
@@ -21,7 +21,7 @@
 import jinja2
 
 from core.controllers import base
-from core.domain.classifier import lda_string_classifier
+from core.domain.classifier import LDAStringClassifier
```

Comment:
A general module like reader.py should not inherit a specific module like LDAStringClassifier. Instead, use the classifier registry (which is much more easily extensible) to get the classifier class. This is similar to what we do for interactions.